spirv_glsl.hpp

```diff
@@ -138,34 +139,48 @@ class CompilerGLSL : public Compiler
 			FeatureCount
 		};
 
+		using FeatureMask = uint32_t;
+		static_assert(sizeof(FeatureMask) * 8u >= FeatureCount, "Mask type needs more bits.");
+
+		static std::initializer_list<Feature> get_feature_dependencies(Feature feature)
+		{
+			switch (feature)
+			{
+			case SubgroupAllEqualT:
+				return { SubgroupBrodcast_First, SubgroupAll_Any_AllEqualBool };
+			case SubgroupElect:
+				return { SubgroupBallotFindLSB_MSB, SubgroupBallot, SubgroupInvocationID };
+			default:
+				return {};
+			}
+		}
+		static FeatureMask get_feature_dependency_mask(Feature feature)
+		{
+			return build_mask(get_feature_dependencies(feature));
+		}
+
 		static bool can_feature_be_implemented_wo_extensions(Feature feature)
 		{
 			const static bool retval[FeatureCount]{
```

Comment:
I think this should be a switch, or get constructed explicitly via
`retval[feature] = value`
so this mapping is not vulnerable to enumerations changing or getting inserted inbetween.